source/Scrapers/BackgroundService.cs

```diff
@@ -1,44 +1,57 @@
 using System;
+using System.Diagnostics;
+using System.Globalization;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Hosting;
 using Polly;
 using Polly.Retry;
+using Serilog;
 
 namespace TeamCityBuildStatsScraper.Scrapers;
 
 //from https://docs.microsoft.com/en-us/dotnet/architecture/microservices/multi-container-microservice-net-applications/background-tasks-with-ihostedservice
 public abstract class BackgroundService : IHostedService, IDisposable
 {
+    protected readonly ILogger Logger;
     Task executingTask;
     readonly CancellationTokenSource stoppingCts = new();
     readonly RetryPolicy retryPolicy;
 
-    protected BackgroundService()
+    protected BackgroundService(ILogger logger)
     {
+        this.Logger = logger;
         retryPolicy = GetRetryPolicy();
     }
     
     protected abstract void Scrape();
 
     async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        stoppingToken.Register(() => Console.WriteLine($"{GetType().Name} background task is starting."));
+        stoppingToken.Register(() => Logger.Information("{TaskName} background task is starting", GetType().Name));
 
         while (!stoppingToken.IsCancellationRequested)
         {
             try
             {
-                retryPolicy.Execute(_ => Scrape(), stoppingToken);
+                retryPolicy.Execute(_ =>
+                {
+                    Logger.Information("Beginning scrape for {TaskName}", GetType().Name);
```

Comment:
This sure looks a lot like a candidate for a trace span at some point in the future.

Do we need to log the 'begin' at Info level? Certainly the 'complete' message has value but I would think the 'begin' is a less important message.

Reply:
> This sure looks a lot like a candidate for a trace span at some point in the future.

👍. One day (soon?) OTEL and logging will be better friends, so this will be less disconnected.

> Do we need to log the 'begin' at Info level? Certainly the 'complete' message has value but I would think the 'begin' is a less important message.

Good call - I've bumped it down.